src/index.js

```diff
@@ -142,149 +137,79 @@ class Node {
     this.swarm.close(callback)
   }
 
-  //
-  // Ping
-  //
-
-  // TODO
-  pingById (id, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-    callback(new Error('not implemented yet'))
+  isOn () {
+    return this.isOnline
   }
 
-  // TODO
-  pingByMultiaddr (maddr, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-    callback(new Error('not implemented yet'))
-  }
-
-  pingByPeerInfo (peerInfo, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
+  ping (peer, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
     callback(null, new Ping(this.swarm, peerInfo))
   }
 
-  //
-  // Dialing methods
-  //
-
-  // TODO
-  dialById (id, protocol, callback) {
-    // NOTE: dialById only works if a previous dial was made. This will
-    // change once we have PeerRouting
-
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-
-    if (typeof protocol === 'function') {
-      callback = protocol
-      protocol = undefined
-    }
-
-    callback(new Error('not implemented yet'))
-  }
-
-  dialByMultiaddr (maddr, protocol, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
-
-    if (typeof protocol === 'function') {
-      callback = protocol
-      protocol = undefined
-    }
-
-    if (typeof maddr === 'string') {
-      maddr = multiaddr(maddr)
-    }
-
-    if (!mafmt.IPFS.matches(maddr.toString())) {
-      return callback(new Error('multiaddr not valid'))
-    }
-
-    const ipfsIdB58String = maddr.stringTuples().filter((tuple) => {
-      if (tuple[0] === IPFS_CODE) {
-        return true
-      }
-    })[0][1]
-
-    let peer
-    try {
-      peer = this.peerBook.getByB58String(ipfsIdB58String)
-    } catch (err) {
-      peer = new PeerInfo(PeerId.createFromB58String(ipfsIdB58String))
-    }
-
-    peer.multiaddr.add(maddr)
-    this.dialByPeerInfo(peer, protocol, callback)
-  }
-
-  dialByPeerInfo (peer, protocol, callback) {
-    assert(this.isOnline, OFFLINE_ERROR_MESSAGE)
+  dial (peer, protocol, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
 
     if (typeof protocol === 'function') {
       callback = protocol
       protocol = undefined
     }
 
-    this.swarm.dial(peer, protocol, (err, conn) => {
+    this.swarm.dial(peerInfo, protocol, (err, conn) => {
       if (err) {
         return callback(err)
       }
-      this.peerBook.put(peer)
+      this.peerBook.put(peerInfo)
       callback(null, conn)
     })
   }
 
-  //
-  // Disconnecting (hangUp) methods
-  //
+  hangUp (peer, callback) {
+    assert(this.isOn(), OFFLINE_ERROR_MESSAGE)
+    const peerInfo = this._getPeerInfo(peer)
 
-  hangUpById (id, callback) {
-    // TODO
-    callback(new Error('not implemented yet'))
+    this.peerBook.removeByB58String(peerInfo.id.toB58String())
```

Comment:
Maybe we should separate these two like go does, so that we have one storage for all details about peers we have ever seen and one storage where we just store the currently connected peers with the address they are connected with.